packages/shared-http-pieces/src/lib/sentry.ts

```diff
@@ -0,0 +1,33 @@
+import { RewriteFrames } from '@sentry/integrations';
+import * as Sentry from '@sentry/node';
+import { fileURLToPath } from 'url';
+
+let initialized = false;
+export function isSentryInitialized() {
+	return initialized;
+}
+
+export function initializeSentry(options: SentryOptions = {}) {
+	Sentry.init({
+		dsn: process.env.SENTRY_DSN,
+		integrations: [
+			new Sentry.Integrations.Console(),
+			new Sentry.Integrations.FunctionToString(),
+			new Sentry.Integrations.LinkedErrors(),
+			new Sentry.Integrations.Modules(),
+			new Sentry.Integrations.Modules(),
+			new Sentry.Integrations.OnUncaughtException(),
+			new Sentry.Integrations.OnUnhandledRejection(),
+			new RewriteFrames({
+				root: options.root ? (typeof options.root === 'string' ? options.root : fileURLToPath(options.root)) : process.cwd()
+			})
+		],
+		...options
+	});
+
+	initialized = true;
```

Comment:
If check if the env var is defined at all before initializing so we don't need to set it in dev

Reply:
The DSN can be defined thru options as well, not just via ENV.

Reply:
wdym through options

Reply:
See the function's parameters and where the object spread is at.

Reply:
This is still not acceptable because this means each bot needs to call the function like so:
```ts
if (isNullishOrEmpty(process.env.SENTRY_DNS)) {
  initializeSentry({ dsn: undefined });
}  else {
  initializeSentry();
}
```
as opposed to just calling `initializeSentry();` with no if check and no parameters because the lib instead handling dev envs.

Reply:
That's what I did with last commit, if the DSN isn't defined via options or env, it's set to false.